tools/classes/AssetsManager.ts

```diff
@@ -0,0 +1,438 @@
+			usedIndexes = new Set<number>();
+		if (cdnAssets.assets) {
+			for (const [index, asset] of cdnAssets.assets) {
+				if (assets.assets.has(asset)) {
+					cdnAssetsInUse.set(index, asset);
+					usedIndexes.add(index);
+				} else result.toBeDeleted.add(asset);
+			}
+		}
+
+		const newAssets = new Set<string>();
+		for (const asset of assets.assets)
+			if (!asset.startsWith(cdnBase)) newAssets.add(asset);
+
+		let index = 0;
+		for (const asset of newAssets) {
+			while (usedIndexes.has(index)) index++;
+			const newAsset = `${this.assetBaseUrl}/${index}${this.getFileExtension(
+				asset
+			)}`;
+			result.toBeUploaded.set(asset, newAsset);
+			usedIndexes.add(index);
+			index++;
+		}
+
+		const missingIndexes = this.findMissing([...usedIndexes]);
+		if (missingIndexes.length) {
+			const cdnAssetsInUseArray = [...cdnAssetsInUse].sort(([a], [b]) => b - a);
+			for (const index of missingIndexes) {
+				const last = cdnAssetsInUseArray.pop();
+				if (!last) break;
+				const [_, asset] = last,
+					newAsset = `${this.assetBaseUrl}/${index}${this.getFileExtension(
+						asset
+					)}`;
+				result.toBeMoved.set(asset, newAsset);
+			}
+		}
+
+		return result;
+	}
+
+	async getCdnAssets(): Promise<{
+		logo: string | false;
+		thumbnail: string | false;
+		assets: Map<number, string> | false;
+	}> {
+		const assets = new Map<number, string>();
+		let assetFound = true,
+			index = 0;
+		while (assetFound) {
+			const asset = await this.doesAssetExistAnyExtension(
+				`${this.assetBaseUrl}/${index}`
+			);
+			if (asset) {
+				assets.set(index, asset);
+				index++;
+			} else {
+				assetFound = false;
+			}
+		}
+
+		return {
+			logo: await this.doesAssetExistAnyExtension(`${this.assetBaseUrl}/logo`),
+			thumbnail: await this.doesAssetExistAnyExtension(
+				`${this.assetBaseUrl}/thumbnail`
+			),
+			assets: assets.size ? assets : false,
+		};
+	}
+
+	canBePut(oldUrl: string, newUrl: string): boolean {
+		return this.getFileExtension(oldUrl) === this.getFileExtension(newUrl);
+	}
+
+	async doesAssetExist(url: string): Promise<boolean> {
+		return got
+			.head(url)
+			.then(() => true)
+			.catch(() => false);
+	}
+
+	async doesAssetExistAnyExtension(url: string): Promise<string | false> {
+		const extensions = [".png", ".jpg", ".jpeg", ".gif", ".webp"];
+		for (const extension of extensions) {
+			const newUrl = `${url}${extension}`;
+			if (await this.doesAssetExist(newUrl)) return newUrl;
+		}
+		return false;
+	}
+
+	async uploadAssets(assets: Map<string, string>) {
+		let errors: string[] = [];
+		for (const [url, newUrl] of assets) {
+			const extension = this.getFileExtension(url);
+			let mimeType: string;
+			switch (extension) {
+				case ".png":
+					mimeType = "image/png";
+					break;
+				case ".jpg":
+				case ".jpeg":
+					mimeType = "image/jpeg";
+					break;
+				case ".gif":
+					mimeType = "image/gif";
+					break;
+				case ".webp":
+					mimeType = "image/webp";
+					break;
```

Comment:
`mime-types` module would be a better fit there and then just disallowing other ones than "image/xyz" or hardcoding some mimes